dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -983,14 +983,21 @@ TRASH_TrashFile(LPCWSTR wszPath)
 }
 
 /*************************************************************************
- * SHUpdateCRecycleBinIcon                                [SHELL32.@]
+ * SHUpdateRecycleBinIcon                                [SHELL32.@]
  *
  * Undocumented
  */
 EXTERN_C HRESULT WINAPI SHUpdateRecycleBinIcon(void)
 {
-    FIXME("stub\n");
+    CComHeapPtr<ITEMIDLIST_ABSOLUTE> pidlRecycleBin;
+    HRESULT hr = SHGetFolderLocation(NULL, CSIDL_BITBUCKET, NULL, 0, &pidlRecycleBin);
+    if (FAILED(hr))
+        return hr;
 
+    // m_pData is referenced directly to ensure that the pointer to
+    // the pidl is given to SHChangeNotify(), not a pointer to the
+    // C++ object that wraps it.
+    SHChangeNotify(SHCNE_UPDATEITEM, SHCNF_IDLIST | SHCNF_FLUSH, pidlRecycleBin.m_pData, NULL);
```

Comment:
Can you point out which function(s) are going to actually change the recycler icon? (here it's not obviously clear since you're sending a shell notification.) -- Also, are you sure that on Windows such notification is being sent when `SHUpdateRecycleBinIcon()` is called?

Reply:
✅